ledger/eval/eval.go

```diff
@@ -788,13 +789,89 @@ func StartEvaluator(l LedgerForEvaluator, hdr bookkeeping.BlockHeader, evalOpts
 		return nil, fmt.Errorf("overflowed subtracting rewards for block %v", hdr.Round)
 	}
 
+	if eval.eligibleForIncentives(prevHeader.Proposer) {
+		incentive, _ := basics.NewPercent(proto.MiningPercent).DivvyAlgos(prevHeader.FeesCollected)
+		total, o := basics.OAddA(incentive, prevHeader.Bonus)
+		if o {
+			return nil, fmt.Errorf("overflowed adding bonus incentive %d %d", incentive, prevHeader.Bonus)
+		}
+
+		sink, err := eval.state.lookup(prevHeader.FeeSink)
+		if err != nil {
+			return nil, err
+		}
+		available := sink.AvailableBalance(&proto)
+		total = basics.MinA(total, available)
+
+		err = eval.state.Move(prevHeader.FeeSink, prevHeader.Proposer, total, nil, nil)
```

Comment:
I'm still thinking about this. Some things are very clear from the header -- the recipient + the _desired_ payment (bonus + % of fees collected).  But a balance lookup of each would be needed to determine eligibility and exact amount.

Reply:
Can we please make it explicit in AD ? 
I have a lot of SQL-based solutions that would be hard to upgrade for incentives if this side effect is not separate from other balance changes that might happen to the winner of the previous proposal.

Reply:
There is no AD for this, since it doesn't appear in a transaction. However, the code I have now (not pushed to github yet because I'm on vacation) I've added a `ProposerPayout` to the block header. That is the final result after adding the rewards together and checking eligibility. So it block `n` has a `ProposerPayout`, block `n+1` moves that much from the feesink to block `n`'s `Proposer` field.